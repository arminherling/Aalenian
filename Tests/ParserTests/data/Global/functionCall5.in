define one() 
{ 
    return 1:i32
}

define inner(x: i32) 
{ 
}

define outer()
{
    inner(one())
}
