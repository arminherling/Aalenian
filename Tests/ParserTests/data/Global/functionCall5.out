FunctionDefinitionStatement: {
    Name: one
    Parameters(0): {
    }
    Block(1): {
        ReturnStatement: {
            Number: 1
            Type: i32
        }
    }
}
FunctionDefinitionStatement: {
    Name: inner
    Parameters(1): {
        Parameter: {
            Name: x
            Type: i32
        }
    }
    Block(0): {
    }
}
FunctionDefinitionStatement: {
    Name: outer
    Parameters(0): {
    }
    Block(1): {
        ExpressionStatement: {
            FunctionCall: {
                Name: inner
                Arguments(1): {
                    Argument: {
                        FunctionCall: {
                            Name: one
                            Arguments(0): {
                            }
                        }
                    }
                }
            }
        }
    }
}
